modin/pandas/test/dataframe/test_join_sort.py

```diff
@@ -166,6 +166,17 @@ def test_join(test_data, test_data2):
         df_equals(modin_join, pandas_join)
 
 
+def test_join_5203():
+    df1 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    df2 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    df3 = pd.DataFrame(np.ones([2, 4]), columns=["a", "b", "c", "d"])
+    with pytest.raises(
```

Comment:
I would suggest to verify that pandas produces the same error as well

Reply:
done